### DISCUSSION TITLE
How do you organize your recipes?

### DISCUSSION THREAD

**Discussion by Benedikt81** [nextcloud/cookbook#1217] [Ideas and discussions]
Currently, we have two ways to organize our recipes:

1. Categories (one per recipe)
2. Keywords (multiple per recipe)

There is no hierarchy in both, whereas in many cookbooks you would have a hierarchy like this:
1. Meat
1.1. Poultry
1.2. Pork
1.3. Beef
2. Vegetarian
2.1. etc.

You can also have a regional categorisation/tagging:
1. East Asian
1.1. Japanese
1.2. Korean
1.3. Chinese

I don't want to suggest new features at the moment, but want to know how you approach the issue.

What I do: I have basic categories which roughly resemble the categories of this [page](https://de.wikibooks.org/wiki/Kochbuch/_Rezeptkategorien). (sorry, it's German, but you can just take a good all-purpose cooking book)

I put the same category in the keywords of the recipe and then add other categories which also apply to the recipe. For example a Lasagne Bolognese gets Pasta, Casserole, Pork, Minced Meat. I also add every ingredient which is significant to the dish as a keyword. So I wouldn't add onions for the Lasagne, but Bechamel Sauce and Tomatoes. Being a spice nerd I also add spices like Basil and Oregano.

In case you wonder, why I do that. I am not fond of wasting food. If I have an ingredient in the fridge which I don't want to spoil, I can open the list of recipes and save the food.

Another keyword I put in íf the food is vegan or vegetarian. If it is vegan, I don't put in vegetarian. I have no need for lactose- or gluten-free keywords, as I have no acquaintances who are on such diets, but it would go in there if needed.

Lastly, I put in one or two geographical keyword. For example, for Pirogi, I put in Eastern European and Polish to resemble a bit of the mentioned hierarchy.

That way, I have a neat selection of recipes by region, category and ingredient. So far, it works well, but the amount of keywords I get is obviously massive.
![image](https://user-images.githubusercontent.com/5584309/191615822-6d2ac3b1-1828-417d-811d-15bb80a76c89.png)

I'm curious how others do it.

**Comment by Joe-ncB** [comment]
Same for me.
I would suggest a third one to separate the ingredients from the keywords -- maybe called ingredients ...
so it is easier to check the fridge and find out what to cook
for the keywords i like to stick with pattern like breakfast, easy, asian, dips, fast, etc

so maybe it can be realised the easy way when the search function will be enhanced
like dragNdrop the keywords to filter instead of typing each character

a big advantage would be in this context to import the ingredients to the keywords with one click
and by a mistaken one to select that one to delete -- at the moment i can only delete from the end to begin
so in worst case i have to insert all again (exept the first one)

another advantage would be that you wouldnt forget a keyword like geografic, type, etc
you will have a better overview of those if the ingredients are apart of they keyword line

at the moment i put a lot of background info of ingredients in the description field -- might be a lot
sometimes i wish i have a separate field for this
or at least a floating text around the picture -- so sometimes there is a big white "nothing" below the picture

btw. more pictures might be nice - smaller ones for intermediate steps - so others can follow easier. ---- maybe in future

i experienced that web sites do not last forever
so i put one or more (for later use) pictures in a local directory
then edit the recipe and and select the local picture (to stay save)

conclusion

when importing a site the system can parse in total its ok
when adding manually good functionality is provided

so when adding a recipe once a week or so no problem
but when creating a really big book from all your notes, links and papers 
it is really time consuming and i wish more automatisation

anyway its a good app and i hope it will grow and last forever

**Reply by Benedikt81** [reply]
>and by a mistaken one to select that one to delete -- at the moment i can only delete from the end to begin
so in worst case i have to insert all again (exept the first one)

You can do that with the dropdown. But you will have scroll a bit if you have many keywords.

![image](https://user-images.githubusercontent.com/5584309/193833186-6e562999-a9d0-4f30-adaf-fdcd40ad8cde.png)

**Reply by seyfeb** [reply]
Actually you can simply start typing to filter. No need to scroll through the whole list ;) Just don't hit enter, otherwise it would create a new keyword (if it does not yet exist)

**Reply by Benedikt81** [reply]
I found that out as well, but it took me a bit. Maybe a more intuitive way would be to click on the keyword to deactivate it. 

- First click: Deactivate
- Second click: Activate (useful if you deactivated by mistake)
- Save recipe: Keyword is removed completely

**Comment by Joe-ncB** [comment]
wow. I just found out when adding manually
you can select all ingredients and/or steps at once and insert them
if they are seperated on the inet site
they will be separated in the app as well

great !!!  much much faster

**Reply by Benedikt81** [reply]
Yes, the instructions or ingredients need to have a new line between each step or ingredient. If I prepare a recipe I usually write it down in a simple text editor like [notepad++ ](https://notepad-plus-plus.org)and then paste it into the app.

**Comment by Joe-ncB** [comment]
hi yes you made it clear. thanks 
edit/delete the keyword works well

@Benedikt81 
sometimes when importing the recipe there is only one big step
there i cut (into buffer) what shall not be in this step 
the add a new empty step and paste it there
and so on

**Comment by Joe-ncB** [comment]
after a big hassle here
i found the documentation an learned how to make a link !!

so all my metadata like geo, history, etc,  goes to a new recipe 
but without picture only the text in the description and the category Knowledge (or if you like background)
then i insert a link to this in the recipes mention this

the advantage is that the orginal recipe keeps short 
and there will be no redundance

**Reply by Benedikt81** [reply]
To put background knowledge into a separate category never occurred to me. Thanks for the suggestion. I think I will do that in the future.

What I already use: Linking recipes like spice blends or doughs from the main recipe. For example my bread recipe links the bread spice blend. With this, I remove redundant information from the database.
![image](https://user-images.githubusercontent.com/5584309/194011725-ee00cda2-edb0-4779-91b8-8a1f544feb3a.png)
